staging/src/k8s.io/apimachinery/pkg/util/net/util.go

```diff
@@ -38,6 +38,20 @@ func IPNetEqual(ipnet1, ipnet2 *net.IPNet) bool {
 	return false
 }
 
+// Returns true if the given err is "connection refused" error.
+func IsConnectionRefused(err error) bool {
```

Comment:
How will we make sure this never breaks? It has some serious assumptions about Golang errors.

Reply:
I guess we can create a unit test for that.

Reply:
https://github.com/golang/go/issues/9383

Reply:
It looks pretty reasonable.  If it fails, it fails in the more reasonable direction